(SIMP-MAINT) Set permissions on /var/log/audit.

zschulte opened the pull request:
Sets the directory permissions to ensure $log_group can traverse the parent directory.

Also sets permissions on /etc/audit/audit.rules.

trevor-vaughan commented:
I'm pretty sure that we had to use the original method to work around issues where the `auditctl` command and startup scripts would cause flapping.

Adding a parameter to the `auditd::config` class would be reasonable. The `log_file_mode` is for the `/var/log/audit/*.log` files themselves instead of the configuration files.

trevor-vaughan commented:
@zschulte Would you like me to make this update, or will you be able to get to it?

zschulte commented:
I had noticed that /etc/audit/audit.rules permission mode was different than the other controlled files in /etc/audit. I attempted to make it identical to those other files.  Would you like me to remove the audit.rules change from this pull in favor of a config_file_mode variable later?

trevor-vaughan commented:
@zschulte Not sure why this just showed up for me but yes, if you could switch over to `log_file_mode` that would be great.

zschulte commented:
@trevor-vaughan I removed the mode change on /etc/audit/audit.rules.  I noticed permission flapping on rotated logs under /var/log/audit so I have also removed the recursive config from /var/log/audit directory control.

trevor-vaughan commented on manifests/config.pp:
```diff
@@ -89,7 +89,7 @@
     owner  => 'root',
     group  => $auditd::log_group,
-    mode   => 'o-rwx'
+    mode   => $log_file_mode,
```
This should have been line 48, right?

zschulte replied:
48 controls the /etc/audit/autid.rules.  This line controls the directory /var/log/audit.  The issue I'm trying to correct is I have a role account in the $auditd::log_group that is unable to traverse /var/log/audit and so cannot access any of the logs even though all the logs are owned by that group.

I did not run into any problems with the /etc/audit/audit.rules, I had just noticed it was similarly controlled by removing other permissions and not setting group permissions.  To keep this pull simple I removed that change.

trevor-vaughan replied:
@zschulte Sorry, I think I may have been a bit confusing.

I think that `log_file_mode` may just be badly named. What I'm getting at is that the auditd configuration files and the auditd logs should not necessarily have the same permissions so we need a second variable.

Something like `config_file_mode` vs `log_file_mode` might do the trick but I'll have to check for backward compatibility issues.

zschulte replied:
I understand.  That is why I removed the change to the config file and left the change to the log directory's permissions.

Currently, this pull will only change the log directory /var/log/audit to be controlled by the variable $log_file_mode.

coreone replied:
@trevor-vaughan it has been a while and we just rebased this to see if we can get this pushed through. Are the changes @zschulte made good, or is there something else that is needed here?

Thanks!

coreone replied:
@trevor-vaughan I just pushed an update (explained in new comment). Can we look into getting this pushed along again? Sorry for the delay.

coreone commented:
Circling back to this. I went through this a little deeper and realized that the permissions for directories was all wrong. I created a new variable to create directory permissions using the same pattern and updated directories to use that new variable.

trevor-vaughan commented on manifests/config.pp:
```diff
@@ -92,7 +97,7 @@
     owner  => 'root',
     group  => $auditd::log_group,
-    mode   => 'o-rwx'
+    mode   => $log_dir_mode,
```
This is the only change that will make a difference from the original implementation.

The intent here was to leave the group permissions alone in case ACLs had been set specifically to allow read access.

What are your thoughts on changing this to `g+rX,o-rwx` if the user is not `root`? We should probably also make this recursive so that the underlying files are changed for reading.

coreone replied:
@trevor-vaughan Does this look correct?

trevor-vaughan replied:
@coreone You can actually get rid of `log_dir_mode` and switch them all back over to `log_file_mode`. Puppet does "the right thing" under the hood for directories.

You'll also need to add `recurse => true` to the `/var/log/audit` resource so that the files are appropriately modified if there's an after-the-fact switch.

trevor-vaughan replied:
@coreone If you're OK with it, I can make the changes real quick  and get this wrapped up.

coreone replied:
If that is set to `recurse`, the next stanza should probably be removed I assume as it should inherit permissions on the log file itself instead of being manually set:

```puppet
file { $auditd::log_file:
  owner => 'root',
  group => $auditd::log_group,
  mode  => $log_file_mode
}
```

trevor-vaughan replied:
Yeah, it would be redundant at that point. Good catch.

coreone replied:
If you want to make any final changes, I'm fine with that. Was just looking through the spec tests and some of those would need to be updated as well.

trevor-vaughan replied:
@coreone Created a PR against your fork
